src/main/java/org/jenkinsci/test/acceptance/docker/fixtures/GitLabContainer.java

```diff
@@ -0,0 +1,217 @@
+package org.jenkinsci.test.acceptance.docker.fixtures;
+
+import org.gitlab4j.api.GitLabApi;
+import org.gitlab4j.api.GitLabApiException;
+import org.gitlab4j.api.GroupApi;
+import org.gitlab4j.api.ProjectApi;
+import org.gitlab4j.api.models.*;
+import org.jenkinsci.test.acceptance.docker.Docker;
+import org.jenkinsci.test.acceptance.docker.DockerContainer;
+import org.jenkinsci.test.acceptance.docker.DockerFixture;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+
+import org.jenkinsci.test.acceptance.po.CapybaraPortingLayer;
+import org.jenkinsci.test.acceptance.utils.ElasticTime;
+
+
+@DockerFixture(id = "gitlab-plugin", ports = {80, 443, 22})
+public class GitLabContainer extends DockerContainer {
+    protected static final String REPO_DIR = "/home/gitlab/gitlabRepo";
+
+    private static final HttpClient client = HttpClient.newBuilder()
+                                                       .followRedirects(HttpClient.Redirect.NORMAL)
+                                                        .connectTimeout(Duration.ofMillis(200))
+                                                       .build();
+
+    private static final ElasticTime time = new ElasticTime();
+
+    public String host() {
+        return ipBound(22);
+    }
+
+    public int port() {
+        return port(22);
+    }
+
+
+    public int webPort() {
+        return port(443);
+    }
+
+    public String webHost() {
+        return ipBound(443);
+    }
+
+    public URL getURL() throws IOException {
+        return new URL("http://" + getIpAddress() + port());
+    }
+
+    public URL getHttpUrl() throws IOException {
+        String url = "http://" + ipBound(80) + ':' + port(80);
+        return new URL(url);
+    }
+
+    /** URL visible from the host. */
+    public String getRepoUrl() {
+        return "ssh://git@" + host() + ":" + port() + REPO_DIR;
+    }
+
+    @Deprecated
```

Comment:
this is new code - why deprecated something that is brand new - just remove the code and fix all callers?

Reply:
deleted it